### PR TITLE
Cater for variable name change in BCRSMatrix of DUNE 2.5

### PR DESCRIPTION
**Pull request by blattms**
One of the variables now has a trailing underline. With this PR we can use DUNE 2.5

### CODE REVIEW

**Comment by atgeirr** [comment]
The code looks good, and I'll merge, but I'll ask for a minor change: could you indent the inner preprocessor #if so that the logic is clear?

File: opm/autodiff/DuneMatrix.hpp [hunk 1]
```diff
@@ -91,7 +91,11 @@ namespace Opm
             nnz = ia[rows];
 
 #if DUNE_VERSION_NEWER(DUNE_ISTL, 2, 3)
+#if DUNE_VERSION_NEWER(DUNE_ISTL, 2, 5)
```
Comment (andlaus) [inline opm/autodiff/DuneMatrix.hpp]: I had/have serious troubles with getting the `DUNE_ISTL_VERSION_*` macros reliably generated by the build system. can this be changed to `DUNE_COMMON`? (that one seems to work better for some reason.)

Comment (atgeirr) [inline opm/autodiff/DuneMatrix.hpp]: Does Dune work with different versions across modules? I would assume not, which should mean that DUNE_VERSION should have been a single thing, not one thing per module. Although perhaps that only applies to the core modules, and the facility is intended to work for user modules, too?

In any case, testing for common or istl should be equivalent, and if they are not that must be a bug!

**Comment by blattms** [comment]
Sure. Done.

**Comment by atgeirr** [comment]
Thanks, will merge when green!

@andlaus if we switch to only using dune-common version info we should do so across the board, and not just in this place. Before you go about replacing all of them I think you should investigate why the DUNE_ISTL version info seems to be unreliable (missing?) in some cases. Also @blattms and @dr-robertk, are there good reasons to use the by-module macros (in OPM) and not just always use the dune-common version?

**Comment by andlaus** [comment]
> Before you go about replacing all of them I think you should investigate why the DUNE_ISTL version info seems to be unreliable (missing?) in some cases.

my current theory is that not all OPM modules that opm-simulators depends on, request these macros. (this could be wrong, though. @akva2: do you want to investigate this directly on my setup?)

**Comment by akva2** [comment]
this is a two-fold problem; the ideal situation is that whenever some module requires the versions this transitions to its children. this is what @dr-robertk added, which we hotfixed for the release. this is broken due to assumptions in the build system which triggers in relation to use of dune-control. the specifics of the problem i have not yet found the time to track down, but it is some crap related to stuff in CONFIG_VARIABLES in the OpmPackage macro not being supposed to have values (ie, #ifdef not #if). it is on my list...

the current state is that unless the top module has
```
opm_need_version_of( "dune-xxx" )
```
in the config hook of the buildsystem, the top module (and thus its parents headers) will not have access to the versions when building. which means top modules need to pay attention to their parents' needs.

**Comment by andlaus** [comment]
> which means top modules need to pay attention to their parents' needs.

I just checked: `opm_need_version_of ("dune-istl")` is featured in the `CMakeLists.txt` of the leaf module (i.e., `opm-simulators`). still, it works only occasionally for me. (I have yet to find out the exact condition.)

**Comment by andlaus** [comment]
maybe because my dune version is installed on the system via the ubuntu packages?

**Comment by akva2** [comment]
it's undoubtedly connected to use of dune-control (or rather, setting the xxx_DIR variables). not blaming dune-control mind you, but that's what triggers the untested path in the build system.
i have never seen issues with it and use system packaged dune.

**Pull request merged**